File: src/main/java/client/threaded/ThreadRenderer.java
package client.threaded;

//TODO: This should be associated with ClientGame and do the drawing.
public class ThreadRenderer extends Thread {

    private void render() {
        /*BufferStrategy bufferstrategy = Game.instance.getBufferStrategy ();

        if (bufferstrategy == null) {
            Game.instance.createBufferStrategy(4);
            return;
        }

        Graphics2D g = (Graphics2D) bufferstrategy.getDrawGraphics();

        g.setColor (Color.BLACK);
        g.fillRect (0, 0, Game.instance.getWidth(), Game.instance.getHeight());
        g.dispose ();
        bufferstrategy.show();
         */
    }

    @Override
    public synchronized void start() {
        super.start();
    }

    @Override
    public void run() {
        //while (Game.isRunning) {
        //    render();
        //}
    }
}

File: src/main/kotlin/client/threaded/ThreadRenderer.kt
package client.threaded

import client.ClientGame
import java.awt.Color
import java.awt.Graphics2D
import java.awt.image.BufferStrategy

//TODO: This should be associated with ClientGame and do the drawing.
// Can be turned into the RenderManager if one is not in use.
class ThreadRenderer : Thread() {

    private fun render() {
        var bufferstrategy : BufferStrategy? = ClientGame.window.bufferStrategy

        if (bufferstrategy == null) {
            ClientGame.window.createBufferStrategy(4)
            return
        }

        var g : Graphics2D = bufferstrategy.drawGraphics as Graphics2D

        bufferstrategy.run {

            g.color = Color.BLACK
            g.fillRect (0, 0, ClientGame.instance.WIDTH, ClientGame.instance.HEIGHT)

            ClientGame.window.draw(g)

            g.dispose ()
            show()
        }

    }

    @Synchronized
    override fun start() {
        super.start()
    }

    override fun run() {
        var bufferstrategy : BufferStrategy? = ClientGame.window.bufferStrategy

        if (bufferstrategy == null) {
            ClientGame.window.createBufferStrategy(4)
            return
        }

        var g : Graphics2D = bufferstrategy.drawGraphics as Graphics2D

        bufferstrategy.run {

            g.color = Color.BLACK
            g.fillRect (0, 0, ClientGame.instance.WIDTH, ClientGame.instance.HEIGHT)

            ClientGame.window.draw(g)

            g.dispose ()
            show()
        }

    }
}
